src/shapes/text.class.js

```diff
@@ -844,13 +854,34 @@
      */
     _setGraphemeOnPath: function(positionInPath, graphemeInfo, startingPoint) {
       var centerPosition = positionInPath + graphemeInfo.kernedWidth / 2,
-          path = this.path;
+          path = this.path, offsetDist = 0;
 
       // we are at currentPositionOnPath. we want to know what point on the path is.
       var info = fabric.util.getPointOnPath(path.path, centerPosition, path.segmentsInfo);
       graphemeInfo.renderLeft = info.x - startingPoint.x;
       graphemeInfo.renderTop = info.y - startingPoint.y;
       graphemeInfo.angle = info.angle + (this.pathSide ===  'right' ? Math.PI : 0);
+
+      if (this.type === 'text' && this.path) {
```

Comment:
why should we check if we have a path and if this is the function `setGraphemeOnPath`? I would also skip the check `this.type === 'text'` since we do not support itext-textbox for now because of editing but anyway this function is going to run anyway.

Reply:
right!